src/core/crypto_tls.c

```diff
@@ -1629,6 +1649,26 @@ QuicCryptoTlsDecodeTransportParameters(
                 "TP: Disable 1-RTT Encryption");
             break;
 
+        case QUIC_TP_ID_VERSION_NEGOTIATION_EXT:
+            if (Length < 5) {
+                QuicTraceEvent(
+                    ConnErrorStatus,
+                    "[conn][%p] ERROR, %u, %s.",
+                    Connection,
+                    Length,
+                    "Invalid length of QUIC_TP_ID_VERSION_NEGOTIATION_EXT");
+                goto Exit;
+            }
+            TransportParams->Flags |= QUIC_TP_FLAG_VERSION_NEGOTIATION;
+            TransportParams->VersionNegotiationInfo = (uint8_t*)TPBuf + Offset; // Review: Allocate and copy?
```

Comment:
Yes, it needs to be copied.

Reply:
The transport parameter doesn't outlive the packet does it? We process it and then we're done?

Reply:
It lives for the lifetime of the connection I believe.